### PR TITLE
chore: add OpenSpec proposal for HTTPS toggle in the game form

### PR DESCRIPTION
**Pull request by CoderCoco**
## Summary

Adds the OpenSpec change `add-https-toggle-to-game-form`. **Proposal only — no application code changes.**

The `replace-alb-with-caddy-sidecar` change turned HTTPS into a per-game flag: set `https = true` in the `game_servers` map and the game's ECS task gains a Caddy sidecar that terminates TLS via Let's Encrypt. Every layer below the UI already carries that flag end to end — the tfvars schema, the zod validator, the IPC and HTTP payloads, the HCL emitter, and the read-only HTTPS field on the game detail page.

The web UI is the one gap. Neither the add-game wizard nor the edit-game form exposes a control for `https`, because the wizard's draft shape (#99) never had a field for it; the edit form works around this by carrying the existing value forward verbatim so a save doesn't wipe it out. The practical consequence is that enabling HTTPS means hand-editing the tfvars object in S3 — exactly the workflow the games UI exists to replace.

## What's in this PR

| File | Contents |
|------|----------|
| `proposal.md` | Scope, impact, and the operator-visible consequences of enabling the flag |
| `design.md` | Design decisions behind the draft plumbing, the control, and the validation rules |
| `specs/game-https-configuration/spec.md` | New capability: how the flag is presented, constrained, explained, and round-tripped |
| `tasks.md` | Seven task groups, validation first |

## Proposed work (for the follow-up implementation PR)

- Add `https` to the shared `WizardDraft` shape so both the wizard and the edit form carry it through their draft → payload round trip, and drop the edit form's carry-forward workaround.
- Render an HTTPS toggle in the Networking step, with an inline amber callout covering all three consequences that a later uncheck cannot undo: 443/80 open stack-wide (the Terraform rule is keyed on "any HTTPS game exists", not per game), the game's raw container port losing public ingress, and first boot performing an ACME issuance that requires `{game}.{hosted_zone_name}` to resolve.
- Port the four Terraform-level `https = true` validation rules from `terraform/aws/variables.tf` into the shared `validateGameServer` business rules, so an invalid configuration is blocked at the form rather than at `terraform validate`. Issues are pathed at `ports[N]` so the offending port row highlights in the Networking step.
- Add `https` round-trip coverage to the tfvars write tests, which currently contain none.

No breaking changes — a game that omits `https` behaves exactly as before, and the flag stays optional and defaults to `false`.

## Test plan

- No code changed, so no test run applies to this PR.
- The proposal's own gates (`tasks.md` §7) require `npm run app:test` and `npm run app:lint` to pass, plus a manual pass through `npm run app:dev`, before the implementation PR merges.
- Task 7.3 notes the implementation PR needs a `Closes #N` line; the tracking issue has not been filed yet.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: openspec/changes/add-https-toggle-to-game-form/design.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Accessibility follows the `Field` helper already in `identity-step.component.tsx`: an associated `<Label htmlFor>`, `aria-describedby` pointing at the callout when it is shown, and `aria-invalid` when a rule is violated.
+
+### D6: Coerce a non-boolean parsed `https` to `false` in the draft
+
+`@cdktf/hcl2json` does not evaluate expressions. A tfvars entry written by hand as `https = length("valheim") > 0 ? true : false` parses back as a *string*, and `TfvarsService.test.ts` documents exactly this. Such an entry already fails `z.boolean()` in `validateGameServer` today, so it cannot be saved through the app at all — with or without this change.
+
+`draftFromGameServer` therefore uses `game.https ?? false`, which yields `false` for a string value rather than a truthy checkbox. This is deliberate and worth stating: the form shows what it can honestly represent, and any save through the form replaces the expression with a literal. That is the only behaviour consistent with a checkbox, and the pre-existing zod rejection means the alternative is not "preserve the expression" but "cannot use the form on this game at all".
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply openspec/changes/add-https-toggle-to-game-form/design.md]: Fixed in `0024d0d`.

Confirmed valid — `??` only substitutes for `null`/`undefined`, so an `hcl2json` expression string would have survived into the draft and rendered the checkbox as checked, contradicting the string-to-`false` test that task 2.2 mandates. Normalised to `game.https === true` in three places: the D6 prose (`design.md:100`), the D6 mapping table (`design.md:52` — a third site your comment did not flag), and task 2.1 (`tasks.md:13`), which now states explicitly that `?? false` is not sufficient.

File: openspec/changes/add-https-toggle-to-game-form/proposal.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+**Unchanged (verified)**
+
+The tfvars types, `gamesWrite.ts` payloads, the preload bridge and `gsd-api.ts`, `api.service.ts`, both games controllers, `GamesWriteService`, `hclEmit`, and the `TfvarsService` write path already handle `https` correctly. No IPC, DTO, or Terraform changes are required.
+
+**Tests**
+
+- `wizard-form.utils.test.ts`, `networking-step.component.test.tsx`, `edit-game-form.component.test.tsx`, `review-step.component.test.tsx`
+- `gameServerValidator.test.ts` — one case per new rule, plus a case proving the rules are inert when `https` is false or absent.
+- `TfvarsService.write.test.ts` — assert `https` survives an update and that flipping it `false → true` rewrites the entry.
```
[the remark this answers is not in the document]
Comment (CoderCoco) [reply openspec/changes/add-https-toggle-to-game-form/proposal.md]: Fixed in `0024d0d`.

Agreed — `https` is optional and most existing tfvars entries omit it, so *omitted → true* is the common enable path and was the one case the write-test plan missed. Added to `proposal.md:53` and as the first case in task 6.1 (`tasks.md:37`), alongside the existing `false → true`, `true → false`, and unrelated-edit cases.

**Comment by CoderCoco** [comment]
## Fixes Applied Successfully

Fixed 3 file(s) based on 2 CodeRabbit feedback item(s).

**Files modified:**
- `openspec/changes/add-https-toggle-to-game-form/design.md`
- `openspec/changes/add-https-toggle-to-game-form/proposal.md`
- `openspec/changes/add-https-toggle-to-game-form/tasks.md`

**Commit:** `0024d0d`

Both items were confirmed valid against the change's own artifacts before applying. The first also had a third occurrence (the D6 mapping table at `design.md:52`) that the original comment did not flag; it was corrected in the same pass for consistency.

`openspec validate add-https-toggle-to-game-form` passes. No application code changed, so the `npm run app:test` / `app:lint` gates in tasks.md §7 remain scoped to the implementation PR.

The latest autofix changes are on the `chore/openspec-https-toggle` branch.

**Comment by CoderCoco** [comment]
Filed the implementation tracking issue: #335. Task 7.3 in `tasks.md` now names it directly instead of carrying a `Closes #N` placeholder (`03c8b42`).

This PR stays proposal-only and does not close #335 — the follow-up implementation PR does.
